web/src/routes/(user)/people/[personId]/+page.svelte

```diff
@@ -91,8 +90,108 @@
       });
     }
   };
+
+  const handleMergeSameFace = async () => {
+    showMergeModal = false;
+    try {
+      await api.personApi.mergePerson({
+        id: personMerge2.id,
+        mergePersonDto: { ids: [personMerge1.id] },
+      });
+
+      people = people.filter((person: PersonResponseDto) => person.id !== personMerge1.id);
+
+      if (personMerge2.name != personName) {
+        ChangeName();
+        invalidateAll();
+      } else {
+        goto(`${AppRoute.PEOPLE}/${personMerge2.id}`, { replaceState: true });
+        notificationController.show({
+          message: 'Merge faces succesfully',
+          type: NotificationType.Info,
+        });
+      }
+    } catch (error) {
+      handleError(error, 'Unable to save name');
+    }
+  };
+
+  const ChangeName = async () => {
+    try {
+      isEditingName = false;
+
+      const { data: updatedPerson } = await api.personApi.updatePerson({
+        id: data.person.id,
+        personUpdateDto: { name: personName },
+      });
+
+      people = people.map((person: PersonResponseDto) => {
+        if (person.id === updatedPerson.id) {
+          return updatedPerson;
+        }
+        return person;
+      });
+
+      notificationController.show({
+        message: 'Change name succesfully',
+        type: NotificationType.Info,
+      });
+    } catch (error) {
+      handleError(error, 'Unable to save name');
+    }
+  };
+
+  const handleNameChange = async (name: string) => {
+    personName = name;
+    let detectnew = false;
+    if (data.person.name != personName) {
+      for (const person of people) {
+        if (person.name === personName && person.id !== data.person.id) {
+          personMerge2 = person;
+          detectnew = true;
+          break;
+        }
+      }
+      if (!detectnew) {
+        await ChangeName();
+        data.person.name = personName;
+        people.map((person: PersonResponseDto) => {
+          if (person.id === data.person.id) {
+            person.name = personName;
+          }
+          return person;
+        });
+      } else {
+        personMerge1 = data.person;
+        potentialMergePeople = people
+          .filter(
+            (person: PersonResponseDto) =>
+              personName === person.name && person.id !== personMerge2.id && person.id !== personMerge1.id,
+          )
+          .slice(0, 3);
+        showMergeModal = true;
```

Comment:
This has like so many if/else for loops, etc. It is very hard to read and understand imo. A few suggestions to improve the readability:
- The for loop is just to find a similar named person. Extract this to a method `findByName(name)`
- Use if + returns where possible to reduce extra indentation and avoid elses
- The goal of components is have some separation of concerns and encapsulate logic. I think you could instead (1) detect if there is a duplicate in this component and (2) pass all the people into the merge component, where it can come up with potential matches, restrict the display to 3, since it is rendering them in that component, etc.